src/read.jl

```diff
@@ -88,122 +88,62 @@ function read_file_header(io::IO)
-    end
-    if isempty(annotations)
-        annotations = nothing
+function read_signal_record!(file::File, signal::Signal, record_index::Int)
+    if isempty(signal.samples)
+        resize!(signal.samples, file.header.record_count * signal.header.samples_per_record)
     end
-    push!(records, record_annotation => annotations)
+    record_start = 1 + (record_index - 1) * signal.header.samples_per_record
+    record_stop = record_index * signal.header.samples_per_record
+    Base.read!(file.io, view(signal.samples, record_start:record_stop))
     return nothing
 end
 
-function read_record_annotation(io::IO)
-    sign = read_sign(io)
-    offset = sign * parse(Float64, String(readuntil(io, 0x14)))
-    if Base.peek(io) !== 0x0
-        events = read_events(io)
-    else
-        events = nothing
-        Base.skip(io, 1)
+function read_signal_record!(file::File, signal::AnnotationsSignal, record_index::Int)
+    io_for_record = IOBuffer(Base.read(file.io, 2 * signal.samples_per_record))
+    tals_for_record = TimestampedAnnotationList[]
+    while !eof(io_for_record) && Base.peek(io_for_record) != 0x00
+        push!(tals_for_record, read_tal(io_for_record))
     end
-    return RecordAnnotation(offset, events)
+    push!(signal.records, tals_for_record)
+    return nothing
 end
 
-function read_timestamp_annotation(io::IO)
-    sign = read_sign(io)
-    raw = readuntil(io, 0x14)
-    timestamp = split(String(raw), '\x15'; keepempty=false)
-    offset = flipsign(parse(Float64, popfirst!(timestamp)), sign)
-    if isempty(timestamp)
-        duration = nothing
-    else
-        duration = parse(Float64, popfirst!(timestamp))
-    end
-    events = read_events(io)
-    events !== nothing || error("No events found for timestamp annotation at offset $offset")
-    return TimestampAnnotation(offset, duration, events)
+function read_tal(io::IO)
+    sign = read_tal_onset_sign(io)
+    bytes = readuntil(io, 0x14)
+    timestamp = split(String(bytes), '\x15'; keepempty=false)
+    onset_in_seconds = flipsign(parse(Float64, timestamp[1]), sign)
+    duration_in_seconds = length(timestamp) == 2 ? parse(Float64, timestamp[2]) : nothing
+    annotations = convert(Vector{String}, split(String(readuntil(io, 0x00)), '\x14'; keepempty=false))
+    return TimestampedAnnotationList(onset_in_seconds, duration_in_seconds, annotations)
 end
 
-function read_sign(io::IO)
+function read_tal_onset_sign(io::IO)
     sign = Base.read(io, UInt8)
-    sign === 0x2b || sign === 0x2d || error("Starting byte of annotation must be '+' or '-'.")
-    return sign === 0x2b ? 1 : -1
-end
-
-function read_events(io::IO)
-    raw = readuntil(io, 0x0)
-    events = split(String(raw), '\x14'; keepempty=false)
-    events = convert(Vector{String}, events)
-    return isempty(events) ? nothing : events
+    sign === 0x2b && return 1
+    sign === 0x2d && return -1
+    error("starting byte of a TAL must be '+' or '-'; found $sign")
```

Comment:
yasguide nitpick: all non-shorthand function defs should use `return` IIRC

Reply:
This does use `return` at the terminating statements